pkg/ccl/sqlccl/backup.go

```diff
@@ -942,12 +947,23 @@ func backupPlanHook(
 
 			for _, d := range targetDescs {
 				if t := d.GetTable(); t != nil {
+					// If we're trying to use a previous backup for this table, ideally it
+					// actually contains this table.
 					if _, ok := tablesInPrev[t.ID]; ok {
 						continue
 					}
+					// This table isn't in the previous backup... maybe was added to a
+					// DB that the previous backup captured?
 					if _, ok := dbsInPrev[t.ParentID]; ok {
 						continue
 					}
+					// Maybe this table is missing from the previous backup because it was
+					// truncated?
+					if len(t.ReplacementOf) > 0 {
+						if _, ok := tablesInPrev[t.ReplacementOf[len(t.ReplacementOf)-1].ID]; ok {
```

Comment:
Looks like you're only interested in the last TRUNCATE. Is this okay?

Reply:
Would be worth adding a comment about that.

Reply:
this block is ensuring that the "previous" backup actually is of the table being incrementally backed up, so it was checking that the prior ID of this table, if it has one, was in it -- I guess we actually should be checking the prior ID as-of that backup though, so that we handle multiple truncations between incremental backups, not the immediate predecessor. I'll change it, and add successive truncates to the test which should catch this case.